### ISSUE TITLE
InstanSeg Image Reader Compatibility for Large OME-TIFF

### ISSUE THREAD

**Issue by jaspreetishar** [instanseg/instanseg#38]
Hi Team,

Thank you for creating InstanSeg—it’s an incredibly helpful tool!

I'm working with a pipeline on an OME-TIFF image that has dimensions of 20294 x 42748 and is approximately 400 MB. However, I’ve encountered the following issue:

When I initialize InstanSeg with `instanseg_brightfield = InstanSeg("brightfield_nuclei", image_reader="tiffslide", verbosity=1)`, the variable `img_pixel_size` receives a `None` value instead of the correct 0.2125. This leads to an error in `instanseg/utils/tiling.py`, specifically:

```python
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

If I switch the image reader to `skimage.io`, `AICSImageIO`, or `bioio`, `img_pixel_size` is correctly set to 0.2125, but I then receive these errors:

```python
NotImplementedError: Image reader AICSImage is not implemented.
NotImplementedError: Image reader bioio is not implemented for whole slide images.
```

Is there a workaround for this? I'm happy to share additional details about the image if that would help.

This is the piece of code I'm using:

```python
from instanseg import InstanSeg
instanseg_brightfield = InstanSeg("brightfield_nuclei", image_reader="bioio", verbosity=1)

labeled_output = instanseg_brightfield.eval(image="../../image.ome.tif",
                                            save_output=True,
                                            save_overlay=True)
```                                                                               

Thank you very much for your time and assistance!

**Comment by ThibautGoldsborough** [comment]
Sorry to hear about this. It looks like InstanSeg tries to switch to whole slide mode (which is still experimental!), you'll get a zarr array as output instead of a tiff label. If you want to force InstanSeg to use eval_medium_image (the image is fully loaded in RAM) then you can increase the
' instanseg_brightfield.medium_image_threshold  = 20294 * 42748 * 10' 

for example. It will be much faster but will require more RAM. If RAM is limiting, then we can suggest you try in QuPath which is much more memory friendly, at least in the short term.

**Comment by jaspreetishar** [comment]
Hi! Thank you for getting back to me. I followed your suggestions and tested on a cloud-based platform with 16 CPUs and 104 GB of memory. Here’s the code I’m using:

```python
from instanseg import InstanSeg
instanseg_brightfield = InstanSeg("brightfield_nuclei", image_reader="bioio", verbosity=1)

instanseg_brightfield.medium_image_threshold = 20294 * 42748 * 10

labeled_output = instanseg_brightfield.eval(image="../../image.ome.tif", save_output=True, save_overlay=True)
```

However, I encountered this error:

```
RuntimeError: quantile() input tensor is too large
```

I also tried different combinations for `medium_image_threshold`, such as `20294 * 42748 * 20`, and so on, but the issue persists. 

Do you have any suggestions for getting past this error? Are there any details about the image that I should share which might be helpful? Thank you again for your help!

**Comment by ThibautGoldsborough** [comment]
This should have been fixed in #40. So just update the repo.

(Alternatively you can give `normalisation_subsampling_factor = 10`) to the eval function and this will perform the quantile calculation on a subset of the image pixels. (results will be very similar but not identical).

**Comment by jaspreetishar** [comment]
Hi there, just wanted to share that the suggestions you provided solved the issue I was facing. Thank you so much for your help - I really appreciate it!
